### PR TITLE
test: recovers unit tests, introduces code coverage, and general lint removal

### PR DESCRIPTION
**Pull request by Gerromie**
# Story

The correct number of test cases that should fail at any time is zero. It's the only way we will be able to know if the code is working as expected. Unfortunately, at the start of this work load, the number of tests failing is almost 90%. This makes it difficult to see if code improvements or new features are effective or causing problems.

Screen Shot 2022-01-04 at 08 10 19

## Changes

Using injection of mocks and additional imports, the existing tests have been
recovered up to a point. They no longer fail because of missing providers
or imports. Many tests are still failing because conditions are not met. This, harder set of problems, will need to be closely evaluated test case by test case.

Screen Shot 2022-01-04 at 08 55 07

After the first iteration, there are still many test case failures, but they are identifying problems with the code itself and, for almost all of the failures, not the test case boilerplate. 

After a lot of effort, the test cases are all passing.

Screen Shot 2022-01-04 at 16 22 47

## Additional Changes

The linter was updated to a more recent version and using `npm run lint` will cause it to run the linter with the cleanup option. This will greatly reduce the remaining lint.

A GitHub action runner now runs the unit tests when the package.json or src files have changed. It will also check and report on lint found in the Angular app.

### CODE REVIEW

**Review by tibco-tdinh — APPROVED** [review]
Sweet! I was gonna try to dive into this myself.

**Comment by Gerromie** [comment]
> Sweet! I was gonna try to dive into this myself.

There's still time! I am working through those test cases I mentioned. I'm currently adding/fixing test cases that fail with the pattern `Cannot read properties of null (reading 'value')` which means that the FormGroup was not defined within the test case.

**Comment by Gerromie** [comment]
Almost all tests are now passing, but those that fail do so intermittently. For example, a refresh of the testing browser window will show issues with the test cases not having data and throwing an exception. These almost always come from a failure to check that the data is present.

Screen Shot 2022-01-04 at 14 48 11

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+name: Project Air UI
```
Comment (torresashjian) [inline .github/workflows/build.yaml:2]: Maybe we can discuss this on Monday, but since we will have effectively 2 different angular apps in the same repository (1 for on-prem, 1 for cloud). Do we need 2 different workflows? Or multiple jobs in the same workflow? Something to think about as we might want to decouple the testing and building.

**Review by torresashjian — APPROVED** [review]
This is great stuff Jeremy! Just a small comment that can be ignored, but we need to discuss in our next meeting for the multiple targets (on-prem, cloud) build, but the rest looks great!
